First draft of reworked user guide (#1482)
skrub-data/skrub: doc/documentation.rst

.. _user_guide:

User Guide
==========

> Skrub is a library that eases the preparation of tabular data in dataframes

Reviewer: ```suggestion
Skrub is a library that eases the machine learning with dataframes
```

.. _user_guide:

User Guide
=============================================================
``skrub`` is a library that eases the preparation of tabular data in dataframes
for machine learning.

Starting from rich, complex data stored in one or several dataframes, it helps
performing the data wrangling necessary to produce a numeric array that is fed
to a machine-learning model. This wrangling comprises joining tables (possibly
with inexact matches), parsing structured data such as datetimes from text,
and extracting numeric features from non-numeric data.

For those tasks, skrub does not replace pandas or polars. Instead, it
leverages the dataframe libraries to provide more high-level building blocks that
perform the data preprocessing steps that are typically needed in a machine learning
pipeline.

> The transformations implemented by ``skrub`` are *stateful*: ``skrub``

Author: A lot of that was in the original user guide and I added it back here, but yes, I agree it may do more harm than good

> The transformations implemented by ``skrub`` are *stateful*: ``skrub``

Reviewer 2: Yes, that's a good opportunity for improving what we currently have
